lib/src/common/disposable_manager.dart

```diff
@@ -112,6 +134,25 @@ abstract class DisposableManagerV3 implements DisposableManagerV2 {
   Completer<T> manageCompleter<T>(Completer<T> completer);
 }
 
+/// Managers for disposable members.
+///
+/// This interface allows consumers to exercise more control over how
+/// disposal is implemented for their classes.
+///
+/// When new management methods are to be added, they should be added
+/// here first, then implemented in [Disposable].
+abstract class DisposableManagerV4 implements DisposableManagerV3 {
+  /// Automatically cancel a stream subscription when this object is disposed.
+  ///
+  /// If the returned stream subscription is managed manually (i.e. canceled
+  /// before disposal of the parent object) [Disposable] will properly manage the
+  /// internal reference.
+  ///
+  /// Neither parameter may not be `null`.
+  StreamSubscription<T> getManagedStreamSubscription<T>(
```

Comment:
Let's talk about the name. In this case the goal is really not so much to "get" a new subscription, but to subscribe to a stream. The subscription is really just a side effect. Can anyone think of a better name, given this?